### PR TITLE
Replace devlog with TYPO3 logger

### PR DESCRIPTION
**Pull request by beatrycze-volk**

### CODE REVIEW

File: Classes/Common/Helper.php [hunk 1]
```diff
@@ -205,35 +206,26 @@ public static function decrypt($encrypted)
...
-                }
-            }
-            $arguments = '(' . implode(', ', $args) . ')';
-            $additionalData = (empty($data) ? false : $data);
-            /** @scrutinizer ignore-deprecated */ \TYPO3\CMS\Core\Utility\GeneralUtility::devLog('[' . $caller . $arguments . '] ' . $message, self::$extKey, $severity, $additionalData);
+        $logger = GeneralUtility::makeInstance(LogManager::class)->getLogger(__CLASS__);
+
+        switch ($severity) {
+            case 0:
+                $logger->info(message);
```
Comment (albig) [inline Classes/Common/Helper.php]: This should be probably `$message`. Below the same error.

**Review by albig — CHANGES_REQUESTED** [review]
Thank you for changing to the logging framework. I found one bug.

Did you notice this change from TYPO3 9.5? This would reduce the again and again MakeInstance for new logger instances:

https://docs.typo3.org/m/typo3/reference-coreapi/master/en-us/ApiOverview/Logging/Logger/Index.html

Could you please add some words about logging to the documentation? There is no "developer" section. But maybe below "System Setup" or "Administrator Manual".

**Review by albig — CHANGES_REQUESTED** [review]
Using the indexer, I get the following exception:

```
typo3 kitodo:reindex -a -p 2 -s dlfCore0

Reindex a collection into database and Solr.
============================================
Uncaught TYPO3 Exception Call to a member function warning() on null
thrown in file /var/www/extensions/dlf/Classes/Common/Document.php
in line 720
```
Could you please have a look at it?

**Comment by beatrycze-volk** [comment]
I have checked and it looks that LoggerAwareTrait doesn't function well with Document class as it has getInstance method. Logger must be initialized here directly in this method after instance is created. I have changed t to previous version and it works.

**Comment by albig** [comment]
> I have checked and it looks that LoggerAwareTrait doesn't function well with Document class as it has getInstance method. Logger must be initialized here directly in this method after instance is created. I have changed t to previous version and it works.

Ok. This is of course stupid but more or less understandable. I'll have another try later.

File: Classes/Common/DocumentList.php [hunk 2]
```diff
@@ -728,7 +733,10 @@ public function __construct(array $elements = [], array $metadata = [])
             // Add initial set of elements to the list.
             $this->elements = $elements;
         }
+
         $this->count = count($this->elements);
+
+        $this->logger = GeneralUtility::makeInstance(LogManager::class)->getLogger(__CLASS__);
```
Comment (albig) [inline Classes/Common/DocumentList.php]: This will fail as LogManager is unknow in this class.

Is this still necessary with LoggerAwareInterface?

**Review by albig — CHANGES_REQUESTED** [review]
Hopefully my last change request ;-)

File: Classes/Common/DocumentList.php (hunk 2 above)
Comment (beatrycze-volk) [reply Classes/Common/DocumentList.php]: No, I have removed it.

**Review by albig — APPROVED** [review]

**Pull request merged**